cds/webapi/C#/MetadataOperations/CDSWebApiServiceExtensions.cs

```diff
@@ -0,0 +1,2312 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PowerApps.Samples.Metadata
+{
+    public static class CDSWebApiServiceExtensions
+    {
+        /// <summary>
+        /// The strong consistency header.
+        /// Strong consistency offers a linearizability guarantee. Linearizability refers to serving requests concurrently.
+        /// The reads are guaranteed to return the most recent committed version of an item.
+        /// Users are always guaranteed to read the latest committed write.
+        /// Use of this header incurs a small overhead and is therefore not suited to high volume workloads.
+        /// </summary>
+        private static readonly Dictionary<string, List<string>> STRONGCONSISTENCYHEADER = new Dictionary<string, List<string>>()
+        {
+            {"Consistency", new List<string> { "Strong" } }
+        };
+
+        /// <summary>
+        /// Adds a solution component to an unmanaged solution.
+        /// </summary>
+        /// <param name="svc">An instance of the CDSWebApiService to extend.</param>
+        /// <param name="componentId">ID of the solution component.</param>
+        /// <param name="componentType">The solution component to add to the unmanaged solution.</param>
+        /// <param name="solutionUniqueName">Unique name of the solution.</param>
+        /// <param name="addRequiredComponents">Indicates whether other solution components that are required by the solution component should also be added to the unmanaged solution.</param>
+        /// <param name="doNotIncludeSubComponents">Indicates whether the subcomponents should be included.</param>
+        /// <param name="includedComponentsSettingsValues">Any settings to be included with the component.</param>
+        /// <returns>The ID of the new solution component record.</returns>
+        public static Guid AddSolutionComponent(this CDSWebApiService svc,
+            Guid componentId,
+            SolutionComponentType componentType,
+            string solutionUniqueName,
+            bool addRequiredComponents,
+            bool? doNotIncludeSubComponents = null,
+            string includedComponentsSettingsValues = null)
+        {
+            Guid solutionComponentId;
+
+            if (!string.IsNullOrEmpty(solutionUniqueName))
+            {
```

Comment:
It's recommended to throw the exception in if clause. Then there would be no else clause.

if (string.IsNullOrEmpty(solutionUniqueName))
            {
throw new Exception("AddSolutionComponent requires the solutionUniqueName parameter value.");
}

Reply:
Done